tokio/src/runtime/driver.rs

```diff
@@ -162,13 +162,18 @@ pub(crate) struct Cfg {
     pub(crate) enable_io: bool,
     pub(crate) enable_time: bool,
     pub(crate) enable_pause_time: bool,
+    pub(crate) start_paused: bool,
 }
 
 impl Driver {
     pub(crate) fn new(cfg: Cfg) -> io::Result<(Self, Resources)> {
         let (io_stack, io_handle, signal_handle) = create_io_stack(cfg.enable_io)?;
 
         let clock = create_clock(cfg.enable_pause_time);
+        if cfg.start_paused {
+            clock.pause();
+        }
+
         let (time_driver, time_handle) =
             create_time_driver(cfg.enable_time, io_stack, clock.clone());
```

Comment:
So this makes it deterministic because `pause` is called _before_ `create_time_driver` sets the start time?

Reply:
That, along with changing the ClockTime constructor to pull start_time out of the clock passed to it rather than going for the global clock (which isn't installed at that point so it falls back to the system clock).